### ISSUE TITLE
Git Hangs Randomly Near Completion

### ISSUE THREAD

**Issue by facekapow** [darlinghq/darling#914]
**Expected Result**
Git should complete successfully without hanging.

**Actual Result**
Git sometimes hangs right when it's about to be finished with index-packing, at around 99%. Happens when cloning Homebrew.

**Steps To Reproduce**
1. Try to clone Homebrew using the instructions at [brew.sh](https://brew.sh).
2. It will sometimes hang when trying to clone Homebrew and/or the core tap.

**System Information**
What system are you using?

| Software | Version |
| --- | --- |
| Linux Kernel | 5.4.0 |
| Darling | 7af86abf751f0cd767491e15aeec51794dabf405 |

**Additional Information**
After seeing the hang and inspecting it with LLDB, here's a little more information. Git's `index-pack` is the one that gets stuck right when it's about to finish. It has an automatic maximum of 4 threads on my system but only has 2 threads still left when the hang occurs: the main thread and a worker thread. The main thread is waiting for the worker thread with `pthread_join` but the worker thread is stuck waiting for a mutex. But for some reason, it seems it never gets woken up for the mutex after everyone else is done with it (it's the only worker thread left, so it can't be locked). Looks like something might be wrong in the LKM... The worst part is that it happens randomly (I Ctrl-C'ed and tried again and it cloned without hanging).

**Comment by facekapow** [comment]
Alright, so at least it doesn't seem to be as bad as it used to be. Just tried clean installs of Homebrew 5 times and none of them hung. So either I fixed it in my last commit (unlikely) or I got lucky. It's also possible that *some* change made it less likely to occur so it's not as bad as it was before; but it still bugs me that it's random issue.

Would really appreciate it if someone else could do some testing on this.
